marlowe/src/Language/Marlowe/Semantics.hs

```diff
@@ -1401,6 +1341,16 @@ instance Eq ReduceEffect where
     _ == _ = False
 
 
+instance Eq FFArg where
+    {-# INLINABLE (==) #-}
+    ArgValueId val1 == ArgValueId val2 = val1 == val2
+    ArgParty val1 == ArgParty val2 = val1 == val2
+    ArgAccountId val1 == ArgAccountId val2 = val1 == val2
+    ArgToken val1 == ArgToken val2 = val1 == val2
+    ArgInteger val1 == ArgInteger val2 = val1 == val2
```

Comment:
Why not have `ArgObservation` and `ArgValue` (the later instead of `ArgInteger`, since it is more general). For completeness, we could also have `ArgLetId`, `ArgChoiceId`, and `ArgPayee`. Probably missing some